staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -1007,9 +1012,13 @@ func decode(codec runtime.Codec, versioner storage.Versioner, value []byte, objP
 }
 
 // appendListItem decodes and appends the object (if it passes filter) to v, which must be a slice.
-func appendListItem(v reflect.Value, data []byte, rev uint64, pred storage.SelectionPredicate, codec runtime.Codec, versioner storage.Versioner, newItemFunc func() runtime.Object) error {
+func appendListItem(
+	v reflect.Value, data []byte, rev uint64, resource string, key []byte,
+	pred storage.SelectionPredicate, codec runtime.Codec, versioner storage.Versioner, newItemFunc func() runtime.Object) error {
 	obj, _, err := codec.Decode(data, nil, newItemFunc())
 	if err != nil {
+		metrics.RecordDecodeError(resource)
+		klog.V(4).Infof("Decoding %s \"%s\" failed: %v", resource, string(key), err)
```

Comment:
can any of the content we're trying to decode leak into `err`? if so, I suspect we shouldn't be logging it (or we could log content of Secret objects, etc)